plugins/lando/lando.plugin.zsh

```diff
@@ -0,0 +1,161 @@
+#!zsh
+#
+# Author: Joshua Bedford
+# URL: github.com/joshuabedford/lando-zsh
+
+#
+# A simple collection of alias functions to enable the use of CLIs within Lando without having to type 'lando'.
+#
+# WARNING: This could conflict with any aliases previously installed. If you have any of the CLIs installed globally (outside of lando)
+#      : The functions *should* enable the ability to not have to type 'lando' before a command by prepending lando for all commands done in the same directory as a .lando.yml file.
+
+
+SITES_DIRECTORY="$HOME/Sites"
+
+CONFIG_FILE=./.lando.yml
+
+# Enable wp command with lando.
+function wp(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    # Run Lando wp
+    lando wp "$@"
+  else
+    # Run System wp
+    command wp "$@"
+  fi
+}
+
+# Enable composer command.
+function composer(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    # Run Lando composer
+    echo "Using 'lando composer'"
+    lando composer "$@"
+  else
+    # Run System composer
+    command composer "$@"
+  fi
+}
+
+# Enable artisan command.
+function artisan(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    # Run Lando artisan
+    lando artisan "$@"
+  else
+    # Run System artisan
+    command artisan "$@"
+  fi
+}
+
+# Enable yarn command for lando if lando file exists in directory.
+function yarn(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    echo "Running Lando yarn...";
+    # Run Lando yarn
+    lando yarn "$@"
+  else
+    echo "Running System yarn...";
+    # Run System yarn
+    command yarn "$@"
+  fi
+}
+
+# Enable npm command for lando if lando file exists in directory.
+function npm(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    echo "Running Lando npm...";
+    # Run Lando NPM
+    lando npm "$@"
+  else
+    echo "Running System npm...";
+    # Run System NPM
+    command npm "$@"
+  fi
+}
+
+# Enable gulp command.
+function gulp(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    echo "Running Lando gulp...";
+    # Run Lando gulp
+    lando gulp "$@"
+  else
+    echo "Running System gulp...";
+    # Run System gulp
+    command gulp "$@"
+  fi
+}
+
+# Enable drush command.
+function drush(){
+
+  if checkForFile $CONFIG_FILE $SITES_DIRECTORY ; then
+    echo "Running Lando drush...";
+    # Run Lando drush
+    lando drush "$@"
+  else
+    echo "Running System drush...";
+    # Run System drush
+    command drush "$@"
+  fi
+}
+
+
+# Check for the file in the current and parent directories.
+# $1: The file to search for (string)
+# $2: The directory to search up to.
+checkForFile(){
+
+  local current_directory="$PWD"
+
+  # Bash is backwards. 0 is true 1 (non-zero) is false.
+  flag="1"
+
+  # Only bother checking for lando within the Sites directory.
+  if [[ ":$PWD:" == *":$2"* ]]; then
```

Comment:
@mcornella The goal there is to check if the current directory is within the sites directory. E.G. Check to see if its a substring. Im not sure that the colons are necessary, and the first star in `*":$2"* is probably unnecessary since the goal is to check the end of the path not the beginning.